src/classes/QueryBuilder.cls

```diff
@@ -2,41 +2,54 @@
 * This file is part of the Nebula Framework project, released under the MIT License.             *
 * See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
 *************************************************************************************************/
-public without sharing class QueryBuilder extends NebulaCore implements IQueryBuilder {
+public without sharing virtual class QueryBuilder extends NebulaCore implements IQueryBuilder {
```

Comment:
at one point we had a test class extending QueryBuilder to inject a mock, but that has since been deprecated!  I will back out that change - looks like we need to do the same!